components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java

```diff
@@ -104,90 +107,103 @@ public static IdentityProvider buildeFederatedIdentityProvider(HttpServletReques
-                            idpRoles.add(new String(value));
-                            paramMap.put(key, new String(value));
-                        } else if (key.startsWith("custom_auth_name")) {
-                            customAuthenticatorNames.add(new String(value));
-                        } else if (key.startsWith("custom_pro_name")) {
-                            proConnectorNames.add(new String(value));
-                        } else if (key.startsWith("cust_auth_prop_")) {
-                            int length = "cust_auth_prop_".length();
-                            String authPropString = new String(key).substring(length);
-                            if (authPropString.indexOf("#") > 0) {
-                                String authName = authPropString.substring(0,
-                                        authPropString.indexOf("#"));
-                                String propName = authPropString.substring(authPropString
-                                        .indexOf("#") + 1);
-                                String propVal = new String(value);
-                                Property prop = new Property();
-                                prop.setName(propName);
-                                prop.setValue(propVal);
-
-                                List<Property> propList = null;
-
-                                if (customAuthenticatorProperties.get(authName) == null) {
-                                    customAuthenticatorProperties.put(authName,
-                                            new ArrayList<Property>());
-                                }
-
-                                propList = customAuthenticatorProperties.get(authName);
-                                propList.add(prop);
-                                customAuthenticatorProperties.put(authName, propList);
+            String idpUUID = StringUtils.EMPTY;
+
+            for (Object item : items) {
+
+                DiskFileItem diskFileItem = (DiskFileItem) item;
+
+                if (diskFileItem != null) {
+
+                    byte[] value = diskFileItem.get();
+                    String key = diskFileItem.getFieldName();
+
+                    if(StringUtils.equals(key, "idpUUID")) {
+                        idpUUID = diskFileItem.getString();
+                    }
+
+                    if ("certFile".equals(key)) {
+                        paramMap.put(key, Base64.encode(value));
+                    } else if ("google_prov_private_key".equals(key)) {
+                        paramMap.put(key, Base64.encode(value));
+                    } else if (key.startsWith("claimrowname_")) {
+                        idpClaims.add(new String(value));
+                        paramMap.put(key, new String(value));
+                    } else if (key.startsWith("rolerowname_")) {
+                        idpRoles.add(new String(value));
+                        paramMap.put(key, new String(value));
```

Comment:
better to assign to a variable and then add/put without creating two string objects